client/core/core.go

```diff
@@ -2039,14 +2056,17 @@ func (c *Core) Login(pw []byte) (*LoginResult, error) {
 					return
 				}
 			}
+			// We wait for connectWalleet to finish before holding the lock as it
+			// may update the deposit address
+			c.walletMtx.Lock()
```

Comment:
The walletMtx was locked outside of the loop over `c.wallets` because it guards access to the `wallets` map.  It needs to go back that way for the loop.

But not the the loop starts goroutines, so the goroutines will fire off, the loop will complete, and `c.walletMtx.Unlock()` will allow each of the goroutines that need to lock it to lock it.